operator/attestor/event_listener.go

```diff
@@ -0,0 +1,60 @@
+package attestor
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type EventListener interface {
+	OnMissedMQBlock()
+	OnBlockMismatch()
+}
+
+const OperatorNamespace = "sffl_operator"
+const AttestorSubsystem = "attestor"
+
+type SelectiveEventListener struct {
+	OnMissedMQBlockCb func()
+	OnBlockMismatchCb func()
+}
+
+func (l *SelectiveEventListener) OnMissedMQBlock() {
+	if l.OnMissedMQBlockCb != nil {
+		l.OnBlockMismatchCb()
+	}
+}
+
+func (l *SelectiveEventListener) OnBlockMismatch() {
+	if l.OnBlockMismatchCb != nil {
+		l.OnBlockMismatchCb()
+	}
+}
+
+func MakeAttestorMetrics(registry *prometheus.Registry) EventListener {
```

Comment:
Currently we don't have more than one instance, but it would be better if all metrics registrations could check for existing metrics, as duplicate metrics directly panic

Reply:
Using non-panicking register now instead. Done